### ISSUE TITLE
Plugins: no information available

### ISSUE THREAD

**Issue by bernd-wechner** [linuxmint/nemo#1293]
The Plugins dialog shows no information on plugins.

![Image of Plugins Dialog](http://i741.photobucket.com/albums/xx55/bwechner/Linux%20MINT/Nemo%20Plugins%20No%20Information.png)

What gives? Is this normal? Any way to diagnose it, or fix it? It's  useful feature and might even help me to know what each plugin actually is!

**Comment by Cos-mo** [comment]
I confirm. The problem exists since Nemo 2.6 (Mint 17.2), when this dialog had been introduced and persists until Nemo 3.0.6 (Mint 18).

Interesting exception: Nemo Fileroller.

**Comment by mtwebster** [comment]
I implemented plugins and modified Nemo Fileroller to supply its own info line (to demonstrate  how to do it) but I haven't had a chance to add info on the remaining ones we maintain.  There's really no way to automatically extract this from somewhere.

**Comment by bernd-wechner** [comment]
Rather odd that this isn't part of the standard plug-in template form day 1. What I hear is that the plugin can offer a description line, but the writer needs to do that. 

Looks like all the extensions live here:

https://github.com/linuxmint/nemo-extensions

and Fileroller returns this here:

```
static GList *
nemo_fr_get_name_and_desc (NemoNameAndDescProvider *provider)
{
    GList *ret = NULL;

    ret = g_list_append (ret, _("Nemo Fileroller:::Allows managing of archives from the context menu"));

    return ret;
}
```

And if I pick an arbitrary one on the list above say nemo-share:

https://github.com/linuxmint/nemo-extensions/blob/master/nemo-share/src/nemo-share.c

It seems not to define nemo_fr_get_name_and_desc, which is the problem?

I mean if I had some clear doc on this, how much work is it to add one to each of these extensions and submit pull requests? 

Just curious. I find this sort of thing shonky. I admit though even if I did it, I wouldn't know what to write for nemo-share as half the problem is lack of documentation, I mean what does it do? I'd have to divine it from the code ... ?

**Comment by bernd-wechner** [comment]
Rather odd that this isn't part of the standard plug-in template form day 1. What I hear is that the plugin can offer a description line, but the writer needs to do that. 

Looks like all the extensions live here:

https://github.com/linuxmint/nemo-extensions

and Fileroller returns this here:

```
static GList *
nemo_fr_get_name_and_desc (NemoNameAndDescProvider *provider)
{
    GList *ret = NULL;

    ret = g_list_append (ret, _("Nemo Fileroller:::Allows managing of archives from the context menu"));

    return ret;
}
```

And if I pick an arbitrary one on the list above say nemo-share:

https://github.com/linuxmint/nemo-extensions/blob/master/nemo-share/src/nemo-share.c

It seems not to define nemo_fr_get_name_and_desc, which is the problem?

I mean if I had some clear doc on this, how much work is it to add one to each of these extensions and submit pull requests? 

Just curious. I find this sort of thing shonky. I admit though even if I did it, I wouldn't know what to write for nemo-share as half the problem is lack of documentation, I mean what does it do? I'd have to divine it from the code ... ?

**Comment by leigh123linux** [comment]
Fix commited for python plugins

https://github.com/linuxmint/nemo-extensions/commit/7031674db859bc1e1543c867530a1b7b38d95ba5

PR for C based plugins

https://github.com/linuxmint/nemo-extensions/pull/181

**Comment by leigh123linux** [comment]
@bernd-wechner 

> I mean if I had some clear doc on this, how much work is it to add one to each of these extensions and submit pull requests?

>Just curious. I find this sort of thing shonky. I admit though even if I did it, I wouldn't know what to write for nemo-share as half the problem is lack of documentation, I mean what does it do? I'd have to divine it from the code ... ?

I used the previous commits to figure it out

For C
https://github.com/linuxmint/nemo-extensions/commit/72031ace3822cc9810a8b3cf7ae665f22d10b71b

For Python

https://github.com/linuxmint/nemo-extensions/commit/7e4bc22672a8b411f6e054497f93729abf4d3a0f

**Comment by mtwebster** [comment]
It's being taken care of.  The ability to turn extensions on and off was something we added recently, it used to be that if an extension was installed, it ran (hence, there was no need for these descriptions,) which was a bit unwieldy to use if you only wanted to use one once in a while, and avoid the overhead cost of them when you weren't using them (some of them can add quite a bit of latency to Nemo)

Note, those extensions that we *don't* maintain ourselves will have to be modified their own developers.  There aren't a lot, but there are a few.  And yes, there is a great deal of divining from the code sometimes - this is mitigated somewhat by being able to look at git history, and familiarity with GLib/GObject programming and patterns.

**Comment by leigh123linux** [comment]
Fixed in 3.2.x
